frontend/src/pages/Player/PlayerPage.tsx

```diff
@@ -152,93 +146,18 @@ const PlayerPage = () => {
 	>(undefined)
 
 	useEffect(() => {
-		if (!session_secure_id) {
+		if (!sessionSecureId) {
 			setShowLeftPanel(true)
 		}
-	}, [session_secure_id, setShowLeftPanel])
+	}, [sessionSecureId, setShowLeftPanel])
 
-	const resizePlayer = useCallback(
-		(replayer: Replayer): boolean => {
-			const width = replayer?.wrapper?.getBoundingClientRect().width
-			const height = replayer?.wrapper?.getBoundingClientRect().height
-			const targetWidth = playerWrapperRef.current?.clientWidth
-			const targetHeight = playerWrapperRef.current?.clientHeight
-			if (!targetWidth || !targetHeight) {
-				return false
-			}
-			const widthScale = (targetWidth - style.PLAYER_PADDING_X) / width
-			const heightScale = (targetHeight - style.PLAYER_PADDING_Y) / height
-			let scale = Math.min(heightScale, widthScale)
-			// If calculated scale is close enough to 1, return to avoid
-			// infinite looping caused by small floating point math differences
-			if (scale >= 0.9999 && scale <= 1.0001) {
-				return false
-			}
+	const { resizeListener, centerColumnResizeListener, controllerWidth } =
+		useResizePlayer(replayer, playerWrapperRef, setScale)
```

Comment:
I believe something with this refactor caused an issue with the player canvas resizing. See this thread for more details: https://highlightcorp.slack.com/archives/C02N2AZS8BV/p1717007998091049